### PR TITLE
Drop support for `node 4` & `node 7`.

### LINKED ISSUE #7213: Remove warning when using npm@5?
Currently, running `ember new foo` on Node 8 will include the following:

<details>
<summary>

```
WARNING: Ember CLI is using the global NPM, but your NPM version has not yet been verified to work with the current Ember CLI release.
```

</summary>

```sh
❯ ember new glim-ber
installing app
  create .editorconfig
  create .ember-cli
  create .eslintrc.js
  create .travis.yml
  create .watchmanconfig
  create README.md
  create app/app.js
  create app/components/.gitkeep
  create app/controllers/.gitkeep
  create app/helpers/.gitkeep
  create app/index.html
  create app/models/.gitkeep
  create app/resolver.js
  create app/router.js
  create app/routes/.gitkeep
  create app/styles/app.css
  create app/templates/application.hbs
  create app/templates/components/.gitkeep
  create config/environment.js
  create config/targets.js
  create ember-cli-build.js
  create .gitignore
  create package.json
  create public/crossdomain.xml
  create public/robots.txt
  create testem.js
  create tests/.eslintrc.js
  create tests/helpers/destroy-app.js
  create tests/helpers/module-for-acceptance.js
  create tests/helpers/resolver.js
  create tests/helpers/start-app.js
  create tests/index.html
  create tests/integration/.gitkeep
  create tests/test-helper.js
  create tests/unit/.gitkeep
  create vendor/.gitkeep
WARNING: Ember CLI is using the global NPM, but your NPM version has not yet been verified to work with the current Ember CLI release.
NPM: Installed dependencies
Successfully initialized git.
```

</details>

npm@5 does seem to work (as long as there is no `npm link`ing involved), should we update the version constraints?

### PR DESCRIPTION
**Pull request by btecu**
Use provided `npm` instead of to version 4. Closes #7213. Closes #7482.

Cleanup some legacy commands used for `npm <= 2`, `node 0.12` and old `yarn` on `travis`.

#7761 got closed accidentally during a bad rebase.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -45,24 +44,8 @@ matrix:
...
-  # prevent the npm loading indicator
-  - npm config --global set spin false
-  # if npm version is less than 3.0.0, upgrade to 3
-  - if [[ $(npm -v | cut -d '.' -f 1) -lt 3 ]]; then npm i -g npm@^3; fi
-  - if [[ $(npm -v | cut -d '.' -f 1) -gt 4 ]]; then npm i -g npm@^4; fi
-
-  # travis currently includes yarn v0.17.8 (20170705)
-  # this isn't new enough for our use of --non-interactive
-  - curl -o- -L https://yarnpkg.com/install.sh | bash
-  - export PATH=$HOME/.yarn/bin:$PATH
```
Comment (Turbo87) [inline .travis.yml]: why did you remove this?

Comment (btecu) [reply .travis.yml]: I thought Travis comes now with a newer `yarn` version that supports `--non-interactive`. Was I wrong?

Comment (Turbo87) [reply .travis.yml]: it's possible, but a) this seems unrelated to the Node 4 changes and b) we might still want to run on the latest yarn version 🤔

Comment (btecu) [reply .travis.yml]: It seemed not necessary - but if the consensus is that it's needed, I'll bring it back.

Comment (twokul) [reply .travis.yml]: let's be conservative and remove it in subsequent PR if needed

Comment (btecu) [reply .travis.yml]: @Turbo87 @twokul Added back

**Comment by Turbo87** [comment]
closing in favor of #7808
